app/scripts/controllers/HomeCtrl.js

```diff
@@ -1,8 +1,9 @@
 (function(){
-  function HomeCtrl(Room, Message, BlocChatCookies, $uibModal) {
+  function HomeCtrl(Room, Message,$cookies, $uibModal) {
 
     this.all = Room.all;
     this.currentRoom = null;
+    this.currentUser = $cookies.get('blocChatCurrentUser');
```

Comment:
Note that this code will run as soon as `HomeCtrl` launches and "cache" the value in `this.currentUser`.  Try clearing your cookies, then setting the user name via modal.  What is `this.currentUser` equal to when you send a message?